inference-engine/src/transformations/src/transformations/common_optimizations/add_fake_quantize_fusion.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/add_fake_quantize_fusion.hpp"
+#include "transformations/utils/utils.hpp"
+
+#include <memory>
+#include <vector>
+
+#include <ngraph/opsets/opset5.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include "itt.hpp"
+
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::AddFakeQuantizeFusion, "AddFakeQuantizeFusion", 0);
+
+ngraph::pass::AddFakeQuantizeFusion::AddFakeQuantizeFusion() {
+    MATCHER_SCOPE(AddFakeQuantizeFusion);
+    auto add_pattern = ngraph::pattern::wrap_type<opset5::Add>({ngraph::pattern::any_input(), ngraph::pattern::wrap_type<opset5::Constant>()},
+                                                               pattern::consumers_count(1));
+    auto fq_pattern = ngraph::pattern::wrap_type<opset5::FakeQuantize>({add_pattern,
+                                                                        ngraph::pattern::any_input(),
+                                                                        ngraph::pattern::any_input(),
+                                                                        ngraph::pattern::any_input(),
+                                                                        ngraph::pattern::any_input()});
+    ngraph::matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        auto fq = std::dynamic_pointer_cast<opset5::FakeQuantize>(m.get_match_root());
+        if (!fq)
+            return false;
+        auto add = std::dynamic_pointer_cast<opset5::Add>(fq->input_value(0).get_node_shared_ptr());
+        if (!add)
+            return false;
```

Comment:
You don't need to cast FQ and Add operations as you don't use operation specific methods from operation classes. And to get nodes please use `m.get_pattern_value_map()`